nemo/collections/common/parts/preprocessing/collections.py

```diff
@@ -489,3 +481,253 @@ def _parse_item(self, line: str, manifest_file: str) -> Dict[str, Any]:
         item = dict(feature_file=item['feature_file'], seq_label=item['seq_label'],)
 
         return item
+
+
+class DiarizationLabel(_Collection):
+    """List of diarization audio-label correspondence with preprocessing."""
+
+    OUTPUT_TYPE = collections.namedtuple(
+        typename='DiarizationLabelEntity',
+        field_names='audio_file duration rttm_file offset target_spks sess_spk_dict clus_spk_digits rttm_spk_digits',
+    )
+
+    def __init__(
+        self,
+        audio_files: List[str],
+        durations: List[float],
+        rttm_files: List[str],
+        offsets: List[float],
+        target_spks_list: List[tuple],
+        sess_spk_dicts: List[Dict],
+        clus_spk_list: List[tuple],
+        rttm_spk_list: List[tuple],
+        max_number: Optional[int] = None,
+        do_sort_by_duration: bool = False,
+        index_by_file_id: bool = False,
+    ):
+        """Instantiates audio-label manifest with filters and preprocessing.
+
+        Args:
+            audio_files:
+                List of audio file paths.
+            durations:
+                List of float durations.
+            rttm_files:
+                List of RTTM files (Groundtruth diarization annotation file).
+            offsets:
+                List of offsets or None.
+            target_spks (tuple):
+                List of tuples containing the two indices of targeted speakers for evaluation.
+                Example: [(0, 1), (1, 2), (0, 2)]
```

Comment:
update the documentation as discussed with list of tuple of tuples for various sessions

Reply:
Modified as discussed

Reply:
updated